pkg/test/mock_client.go

```diff
@@ -0,0 +1,53 @@
+package test
```

Comment:
I think you can use this: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/fake/client.go

Reply:
The fake client doesn't allow me to mock errors during update or get

Reply:
This currently has no value anywhere else. IMO, we should either add something like [this](https://github.com/kiegroup/kie-cloud-operator/blob/master/pkg/controller/kieapp/test/mock_service.go) that can do what you want as well as others, or localize this functionality to your unit tests.

Reply:
Do you mean move the mock_service to this library?

Reply:
Right, that's one option, given that mock_service would allow you to do what you're doing in these unit tests.

My concern is that right now, we have a utility in the top level pkg/test that is very narrow in use and only serves to fill the need of a specific test.

Reply:
The mock_client can be also used in the writer and reader tests. Actually I think it is more useful for testing methods like writer#AddResources or writer#UpdateResources where the error path is not being tested